### DISCUSSION TITLE
Allow uploaded artifact file name to be different as distribution file name itself

### DISCUSSION THREAD

**Discussion by marceloverdijk** [jreleaser/jreleaser#1623] [Q&A]
I'm currently releasing artifacts to Github like:

```
jreleaser {
    project {
        ..
    }
    files {
        artifact {
            setPath(layout.buildDirectory.file("distributions/${projectName}-csv-{{projectVersion}}.zip").get().asFile.path)
        }
        artifact {
            setPath(layout.buildDirectory.file("distributions/${projectName}-json-{{projectVersion}}.zip").get().asFile.path)
        }
```

It uses the files produced by the Gradle distribution plugin, and as you can see it includes the `{{projectVersion}}`.

It would be nice to override the name of the file that is uploaded to the Github releases.

Rationale: I would like to upload files without the `-{{projectVersion}}` part.
This so it would be easily to download latest assets from Github using: 
www.github.com/my-repo/my-project/releases/latest/download/my-project-name-csv.zip
www.github.com/my-repo/my-project/releases/latest/download/my-project-name-json.zip

The problem is the Gradle distributions plugin always seems to include the vesion in the dist file, so I'm trying other ways to upload it with jreleaser plugin without version in the file name.

**Comment by aalmiray** [comment]
Have you tried using the `transform` property? https://jreleaser.org/guide/latest/concepts/artifacts.html JReleaser may rename the artifact regardless of the filename produced by the build.

Here's an example:

https://github.com/jreleaser/jreleaser/blob/67838bb88a7b9ab2bb008611c20bfb901b063d2d/jreleaser.yml#L463-L475

Your config could look like

```kotlin
        artifact {
            setPath(layout.buildDirectory.file("distributions/${projectName}-csv-{{projectVersion}}.zip").get().asFile.path)
            setTransform("${projectName}-csv.zip")
        }
```

**Comment by marceloverdijk** [comment]
Ah I see, but I now understand why my IDE was not giving me that option,
as when adding `setTransform(..)` it gives me: `Unresolved reference: setTransform`

https://github.com/jreleaser/jreleaser/blob/main/plugins/jreleaser-gradle-plugin/src/main/groovy/org/jreleaser/gradle/plugin/dsl/common/Artifact.groovy#L30-L38

There seems no `setTransform` method in the `Artifact` interface, like there is one for `setPath`.

**Comment by aalmiray** [comment]
Ah, that is correct. You must use direct property setter such as `transform.set("...")`

**Comment by marceloverdijk** [comment]
Oh indeed. Sometimes a bit of puzzling with Gradle-Kotlin.. Thx 🥳

**Reply by aalmiray** [reply]
Supposedly you should also be able to write `transform = "..."` like in Groovy.
